// system/apis.js
/**
 * New node file
 */

var api_version = '0.1';
var get_api_version = function() { return api_verion;};
var set_api_version = function(new_api_version) {  api_version=new_api_version;};

var endpoint_url = 'api.positronics.io';
var get_base_url = function() { return endpoint_url+'/'+api_version;};
var location_id = 'location_id';
var items = 'items';
var modifiers = 'modifiers';
var modifier_id = 'modifier_id';
var modifier_groups = 'modifier_groups';
var modifier_group_id = 'modifier_group_id';
var menu = 'menu';
var categories = 'categories';
var category_id = 'category_id';
var menu_item_id = 'menu_item_id';
var tickets = 'tickets';
var ticket_id = 'ticket_id'
var ticket_item_id = 'ticket_item_id';
var ticket_item_modifier_id = 'ticket_item_modofier_id';
var tables = 'tables';
var table_id = 'table_id';
var employees = 'employees';
var location_id = 'location_id';
var employee_id = 'employee_id';
var order_types = 'order_types';
var order_type_id = 'order_type_id';
var tender_types = 'tender_types';
var tender_type_id = 'tender_type_id';
var revenue_centers = 'revenue_centers';
var revenue_center_id = 'revenue_center_id';
var payments = 'payments';
var payment_id = 'payment_id';



var apis = {

		tables					: [tables,table_id,],
		employees   			: [employees,employee_id,],
		order_types 			: [order_types,order_type_id,],
		tender_types 			: [tender_types,tender_type_id,],		
		revenue_centers 		: [revenue_centers,revenue_center_id,],		
		tickets					: [tickets,ticket_id,items,ticket_item_id,modifiers,ticket_item_modifier_id,],		
		payments				: [payments,payment_id],		
		//...
		//...order is important
		//...
		menu 					: [menu,],
		menu_categories 		: [menu,category_id,],
		menu_item				: [menu,menu_item_id,modifier_groups,modifier_group_id,],
		menu_modifiers			: [menu,modifier_id,],
};


var urls = function () {

	urls = {locations : '/locations/location_id'};
	for(api in apis) {
		var _urls = [];
		var url = '/locations/location_id';
		for(i in apis[api]) {
			url += '/'+apis[api][i];
 			if(apis[api][i]!=undefined) {
				urls[apis[api][i]] = url;
			}
		}
	}
	return urls;

}();

module.exports = function() {
	return {
		//...
		get_api_version: get_api_version,
		set_api_version: set_api_version,
		get_base_url: get_base_url,
		//...
		urls: urls,
		//...
		location_id: location_id,
		//...
		ticket_id: ticket_id,
		ticket_item_id: ticket_item_id,
		items: items,
		tickets: tickets,
		modifiers: modifiers,
		modifier_id: modifier_id,
		//...
		table_id: table_id,
	};
	
};
