// system/tickets.js
/**
 * New node file
 */
var apis = require('./apis.js')();

var gen_ticket_args = function (p_location_id,p_ticket_id,p_items,p_ticket_item_id,p_modifiers,p_modifier_id) {
	var args = {
			location_id:p_location_id,
			ticket_id:p_ticket_id,
			items:p_items,
			ticket_item_id:p_ticket_item_id,
			modifiers:p_modifiers,
			modifier_id:p_modifier_id,
	};
	return args;
}

//...
//...tickets
//...
function get_location_tickets_url(p_location_id) {
	var the_url = apis.get_base_location_url(p_location_id)+apis.urls.tickets
	return the_url;
}

//...
//...ticket
//...
function get_location_ticket_id_url(p_args) {
	if(p_args.ticket_id==undefined) {
		throw new Error('missing ticket_id');
	}
	var the_url = apis.get_base_location_url(p_args.location_id)+apis.urls.ticket_id;
	the_url = the_url.replace(apis.ticket_id,p_args.ticket_id);
	return the_url;
}

function get_location_ticket_items_url(p_args) {
	the_url = get_location_ticket_id_url(p_args);
	if(p_args.items!=undefined) {
		the_url += '/'+apis.items;		
	}
	return the_url;
}

function get_location_ticket_item_id_url(p_args) {
	the_url = get_location_ticket_items_url(p_args);
	if(p_args.ticket_item_id!=undefined) {
		if(p_args.items==undefined) {
			throw new Error('missing items keywords');
		}
		the_url += '/'+p_args.ticket_item_id;		
	}
	return the_url;
}

function get_location_ticket_item_id_modifiers_url(p_args) {
	the_url = get_location_ticket_item_id_url(p_args);
	if(p_args.modifiers!=undefined) {
		if(p_args.ticket_item_id==undefined) {
			throw new Error('missing ticket_item_id keyword');
		}
		the_url += '/'+apis.modifiers;		
	}
	return the_url;
}

function get_location_ticket_item_id_modifier_id_url(p_args) {
	the_url = get_location_ticket_item_id_modifiers_url(p_args);
	if(p_args.modifier_id!=undefined) {
		if(p_args.modifiers==undefined) {
			throw new Error('missing modifiers keyword');
		}
		the_url += '/'+p_args.modifier_id;
	}
	return the_url;
}


var get_location_ticket_url = function(p_args) {
	return get_location_ticket_item_id_modifier_id_url(p_args);
	
}

module.exports = function() {

	return {
		get_location_tickets_url: get_location_tickets_url,
		gen_ticket_args: gen_ticket_args,
		get_location_ticket_url: get_location_ticket_url,
		
	};
}
