// system/urls.js
/**
 * New node file
 */
var api_version = '0.1';
var endpoint_url = 'api.positronics.io';

var locations = '/locations';
var menu = 'menu';
var tickets = 'tickets';
var tables = 'tables';
var employees = 'employees';
var location_id = '{location_id}';
var table_id = '{table_id}';
var employee_id = '{employe_id}';
var ticket_id = '{ticket_id}'

var location_tickets = locations+'/'+location_id+'/'+tickets;
var location_menu = locations+'/'+location_id+'/'+menu;
var location_tables = locations+'/'+location_id+'/'+tables;
var location_table = locations+'/'+location_id+'/'+tables+'/'+table_id;
var location_employees = locations+'/'+location_id+'/'+employees;
var location_employee = locations+'/'+location_id+'/'+employees+'/'+employee_id;

var get_api_version = function() { return api_verion;};
var set_api_version = function(new_api_version) {  api_version=new_api_version;};
var get_base_url = function() { return endpoint_url+'/'+api_version;};

var get_locations_url = function() {
	return get_base_url()+locations;
}
var get_location_tickets_url = function(p_location_id) { 
	var location_tickets_for_location_id = get_base_url()+location_tickets.replace(location_id,p_location_id);
	return location_tickets_for_location_id;
	
}
var get_location_ticket_url = function(p_location_id,p_ticket_id) { 
	var location_ticket_for_location_id = get_base_url()+location_tickets.replace(location_id,p_location_id);
	location_ticket_for_location_id = location_ticket_for_location_id.replace(ticket_id,p_ticket_id);
	return location_ticket_for_location_id;
}

var get_location_menu_url = function(p_location_id) { 
	var location_menu_for_location_id = get_base_url()+location_menu.replace(location_id,p_location_id);
	return location_menu_for_location_id;
}

var get_location_tables_url = function(p_location_id) { 
	var location_tables_for_location_id = get_base_url()+location_tables.replace(location_id,p_location_id);
	return location_tables_for_location_id;
}

var get_location_table_url = function(p_location_id,p_table_id) { 
	var location_table_for_location_id = get_base_url()+location_table.replace(location_id,p_location_id);
	location_table_for_location_id = location_table_for_location_id.replace(table_id,p_table_id);
	return location_table_for_location_id;
}

var get_location_employees_url = function(p_location_id) { 
	var location_employees_for_location_id = get_base_url()+location_employees.replace(location_id,p_location_id);
	return location_employees_for_location_id;
}

var get_location_employee_url = function(p_location_id,p_employee_id) { 
	var location_employee_for_location_id = get_base_url()+location_employee.replace(location_id,p_location_id);
	location_employee_for_location_id = location_employee_for_location_id.replace(employee_id,p_employee_id);
	return location_employee_for_location_id;
}

exports.get_api_version = get_api_version;
exports.set_api_version = set_api_version;
exports.get_base_url = get_base_url;
exports.get_locations_url = get_locations_url;
exports.get_location_tickets_url = get_location_tickets_url;
exports.get_location_ticket_url = get_location_ticket_url;
exports.get_location_menu_url = get_location_menu_url;
exports.get_location_tables_url = get_location_tables_url;
exports.get_location_table_url = get_location_table_url;
exports.get_location_employees_url = get_location_employees_url;
exports.get_location_employee_url = get_location_employee_url;
