// system/urls.js





var api_version = '0.1';
var endpoint_url = 'api.positronics.io';


var get_api_version = function() { return api_verion;};
var set_api_version = function(new_api_version) {  api_version=new_api_version;};
var get_base_url = function() { return endpoint_url+'/'+api_version;};





var apis = require('./apis.js')();

var gen_ticket_args = function (p_location_id,p_ticket_id,p_items,p_ticket_item_id,p_modifiers,p_modifier_id) {
	var args = {
			location_id:p_location_id,
			ticket_id:p_ticket_id,
			items:p_items,
			ticket_item_id:p_ticket_item_id,
			modifiers:p_modifiers,
			modifier_id:p_modifier_id,
			
	};
	return args;
}

//...
//...tickets
//...
function get_location_tickets_url(p_location_id) {
	var the_url = get_base_url()+apis.urls.tickets
	the_url = the_url.replace(apis.location_id,p_location_id);
	return the_url;
}

//...
//...ticket
//...
function get_location_ticket_id_url(p_args) {
	if(p_args.location_id==undefined) {
		throw new Error('missing location_id');
	}
	if(p_args.ticket_id==undefined) {
		throw new Error('missing ticket_id');
	}
	var the_url = get_base_url()+apis.urls.ticket_id;
	the_url = the_url.replace(apis.location_id,p_args.location_id);
	the_url = the_url.replace(apis.ticket_id,p_args.ticket_id);
	return the_url;
}

function get_location_ticket_items_url(p_args) {
	the_url = get_location_ticket_id_url(p_args);
	if(p_args.items!=undefined) {
		the_url += '/'+apis.items;		
	}
	return the_url;
}

function get_location_ticket_item_id_url(p_args) {
	the_url = get_location_ticket_items_url(p_args);
	if(p_args.ticket_item_id!=undefined) {
		if(p_args.items==undefined) {
			throw new Error('missing items keywords');
		}
		the_url += '/'+p_args.ticket_item_id;		
	}
	return the_url;
}

function get_location_ticket_item_id_modifiers_url(p_args) {
	the_url = get_location_ticket_item_id_url(p_args);
	if(p_args.modifiers!=undefined) {
		if(p_argsticket_item_id==undefined) {
			throw new Error('missing ticket_item_id keyword');
		}
		the_url += '/'+modifiers;		
	}
	return the_url;
}

function get_location_ticket_item_id_modifier_id_url(p_args) {
	the_url = get_location_ticket_item_id_modifiers_url(p_args);
	if(p_args.modifier_id!=undefined) {
		if(p_args.modifiers==undefined) {
			throw new Error('missing modifiers keyword');
		}
		the_url += '/'+apis.modifier_id;
	}
	return the_url;
}


var get_location_ticket_url = function(p_args) {
	return get_location_ticket_item_id_modifier_id_url(p_args);
	
}



exports.get_api_version = get_api_version;
exports.set_api_version = set_api_version;

exports.get_base_url = get_base_url;

exports.get_location_tickets_url = get_location_tickets_url;

exports.gen_ticket_args = gen_ticket_args;
exports.get_location_ticket_url = get_location_ticket_url;
